Initial spec doc contribution for Jakarta EE

OndroMih opened the pull request:
Initial modifications by @tivrfoa of the donated document for Jakarta EE. This is a follow-up for #267, which was closed before merged.

This pull request builds on #273. After #273, this PR will only contain changes made in the donated document to make it easier to review.

(This PR replaces #274 which wasn't synchronized with master and didn't show only diffs in the spec file)

hussainnm commented on spec/src/main/asciidoc/messaging-spec.adoc:
```diff
@@ -250,39 +250,39 @@ destination of messages it is sending and the source of messages it
+=== Jakarta Messaging APIs
 
-For historical reasons JMS offers four alternative sets of interfaces
+For historical reasons Jakarta Messaging offers three alternative sets of interfaces
```
Why change to three - (1.0 = 2, 1.1 = 1, 2.0 = 1 = four)

OndroMih replied:
I reverted this back to "four"

OndroMih commented:
@hussainnm, I'm going to fix the wrong references to JCA, EJB and JTA. 

I'm not sure whether to change references to "EJB Container" because the spec in the master branch of Jakarta Enterprise Beans currently still defines it as "EJB Container". I also asked about it in the Enterprise Beans github issue here: https://github.com/eclipse-ee4j/ejb-api/issues/99#issuecomment-625330813

OndroMih commented:
The Enterprise Beans PR https://github.com/eclipse-ee4j/ejb-api/pull/96/files#diff-adade127f19060d71cb75f896da4c40cR89 proposes to change "EJB Container" to "Enterprise Beans Container". I'll replace "EJB Container" to match that.

OndroMih commented:
@hussainnm, I've replaced all references to JCA, EJB and JTA:

* `Jakarta EE Connector Architecture` -> `Jakarta Connectors`
* `EJB` and `Enterprise JavaBeans` -> Jakarta Enterprise Beans
* `EJB Container` -> `Enterprise Beans container`
* `JTA` -> `Jakarta Transactions`
* `JTA transaction` -> `Jakarta Transaction`

I also removed the scope section which became redundant. It caused that the section numbers were shifted and it was there only as a stub before the Java EE document was donated.

hussainnm commented on spec/src/main/asciidoc/messaging-spec.adoc:
```diff
@@ -2800,25 +2796,25 @@ always has a current transaction within which its work is done.
 Jakarta Messaging does not require that a provider support distributed transactions;
 however, it does define that if a provider supplies this support it
-should be done via the JTA XAResource API.
+should be done via the Jakarta Transactions XAResource API.
```
XAResource API is still under JDK and not part of Jakarta Transactions.

OndroMih replied:
So what do you suggest? Revert this back to JTA XAResource API?

OndroMih replied:
Change yo Java XA Resource API as suggested.

hussainnm commented on spec/src/main/asciidoc/messaging-spec.adoc:
```diff
@@ -4797,7 +4790,7 @@ close method when the object is not longer needed.
 
-Applications may declare a field of type javax.jms.JMSContext and
+Applications may declare a field of type jakarta.jms.JMSContext and
 annotate it with the javax.inject.Inject annotation:
```
```suggestion
annotate it with the jakarta.inject.Inject annotation:
```

hussainnm commented on spec/src/main/asciidoc/messaging-spec.adoc:
```diff
@@ -4546,55 +4541,54 @@ Table 11‑1 Relationship of optional interfaces in domains
 include the following:
 
 * Support for distributed transactions which are demarcated either
 programmatically, using methods on javax.transaction.UserTransaction, or
```
```suggestion
programmatically, using methods on jakarta.transaction.UserTransaction, or
```

hussainnm commented on spec/src/main/asciidoc/messaging-spec.adoc:
```diff
@@ -4607,88 +4601,87 @@ the first JMSContext already contains a connection and session and this
-This applies irrespective of whether the JTA transaction is demarcated +
+This applies irrespective of whether the Jakarta transaction is demarcated +
 automatically by the container or programmatically using methods on +
 javax.transaction.UserTransaction.
```
```suggestion
jakarta.transaction.UserTransaction.
```

hussainnm commented on spec/src/main/asciidoc/messaging-spec.adoc:
```diff
@@ -642,33 +638,33 @@ waiting for them. A message-based application triggering mechanism is
+Jakarta Messaging provides the JMSReplyTo message header field for specifying the
 Destination where a reply to a message should be sent. The
 JMSCorrelationID header field of the reply can be used to reference the
 original request. See Section “” for more information.
```
There are total 9 missing references like this. You can search for empty quotes in the document.

OndroMih replied:
There were more, I found 16 missing references. I copied the references from the JMS 2.0 specification. I also checked that the section numbers are correct and they are.

hussainnm commented on spec/src/main/asciidoc/messaging-spec.adoc:
```diff
@@ -102,84 +99,84 @@ _topic_. The client that sends the message _publishes_ it to a specific
-The JMS API is designed to be suitable for use by both Java client
+The Jakarta Messaging API is designed to be suitable for use by both Java client
 applications using the Java™ Platform, Standard Edition (Java SE), and
 Java middle-tier services using the Java™ Platform, Enterprise Edition
```
2 occurrences of Java™ Platform, Enterprise Edition needs to be changed to Jakarta EE Platform.

OndroMih replied:
I've fixed those references. 

I also removed references to specific versions of Jakarta EE platform or Enterprise Beans as they obsolete with every new version. They were confusing anyway because they referred to an older version of Java EE because the JMS spec was finalized before the version of Java EE spec that included it. If we don't use the exact version, it should be implicitly the Jakarta EE version that includes this version of Jakarta Messaging.

OndroMih commented:
No problem with raising more issues as you continue reviewing the document, @hussainnm. I'm glad that you're so thorough. Is it OK if I keep fixing issues you report as soon as possible or if I wait until you're finished with your review? What's more convenient for you?

hussainnm commented on spec/src/main/asciidoc/messaging-spec.adoc:
```diff
@@ -1245,18 +1241,18 @@ properties are undefined.
 ==== Provider-specific properties
 
-JMS reserves the ‘JMS___<vendor_name>__’ property name prefix for
+Jakarta Messaging reserves the ‘JMS___<vendor_name>__’ property name prefix for
```
There is a problem here with the underscore missing between JMS and <vendor_name>. The missing underscore is showing up on the next line
This is formatting issue.

![image](https://user-images.githubusercontent.com/16149023/81455477-f5a73580-91ac-11ea-9d67-79c9e2fe1d17.png)

![image](https://user-images.githubusercontent.com/16149023/81455530-26876a80-91ad-11ea-8082-3bab45c3d23e.png)

OndroMih replied:
Fixed.

hussainnm commented on spec/src/main/asciidoc/messaging-spec.adoc:
```diff
@@ -3102,7 +3101,7 @@ send(Destination destination, Message message)
 
 ____
```
Quote block is not required
![image](https://user-images.githubusercontent.com/16149023/81456641-ba0e6a80-91b0-11ea-9141-3a9fa42e842c.png)

OndroMih replied:
Fixed.

OndroMih commented:
Thanks for more comments, @hussainnm.

I believe I've fixed all except I didn't change keywords in the text as it would be too much work now and also would pollute the diff for this PR. I may raise a separate PR after this is merged.

Here's what I've fixed today:

* missing underscore in Message selector syntax and provider-specific properties
* Removed headers for Table 8-1 and 8-2
* Fixed all code blocks
Turned chapter 16 into Appendix (existing references to Appendix should now be OK)

hussainnm commented on spec/src/main/asciidoc/messaging-spec.adoc:
```diff
@@ -5860,60 +5817,106 @@ Note that sendMessageNew does not need to throw JMSException.
+        context.createProducer().send(dataQueue,body);
+    }
+}
 
```
Missing closing block (----), this is messing rest of the document.

OndroMih replied:
I was afraid this could happen, I should've doublechecked. This is fixed and I also fixed it in one more place which Asciidoctor reported as missing ending element.

OndroMih commented:
> Line 6052 and 6074 is missing closing block.
Fixed. There were a few more which I discovered using a regex search.

arjantijms commented:
Amazing work everyone! Thanks a bunch :) 

I'll merge this now. If there are additional changes needed they can be done in follow-up PRs.
